webpack/assets/javascripts/react_app/common/forceSingleton.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Force a single instance to protect from code duplication.
+ *
+ * WARNING: Code duplications happen because of an issue with the build process,
+ *          so this method might be removed once the issue would be fixed.
+ *          See: https://projects.theforeman.org/issues/27195
+ *
+ * @param  {String}   key    A unique-key to save the instance.
+ * @param  {Function} create A function to create an instance.
+ * @return {*}               Single Instance,
+ *                           returned by the create method or from the cache.
+ */
+const forceSingleton = (key, create) => {
```

Comment:
I guess in many cases the default would be an empty object, would it make sense to use:
```suggestion
const forceSingleton = (key, create = () => ({})) => {
```
and only pass the second attribute if needed?

Reply:
IMO it's a hack and I'm not sure about its place once we will fix the build system so
maybe we shouldn't make it too friendly?